### PR TITLE
optimizer refactor

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

**Comment by tqchen** [comment]
@piiswrong can you fix the lint error? We should aim to bring quick refactors in

**Comment by tqchen** [comment]
Is the name lr_mult commonly used in learning rate scaling? 

Please do the following things
- Add a few unit testcases on the new attribute function to ensure setter/getter works correctly.

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -114,15 +120,51 @@ def create_state(self, index, weight):
...
-        self.lr_scale = args_lrscale.copy()
+        self.lr_mult = {}
+        if self.sym is not None:
+            attr = self.sym.list_attr()
+            for k, v in attr.items():
+                if k.endswith('_lr_mult'):
+                    self.lr_mult[k[:-len('_lr_mult')]] = float(v)
+        self.lr_mult.update(args_lr_mult)
+
+    def set_wd_mult(self, args_wd_mult):
```
Comment (tqchen) [inline python/mxnet/optimizer.py]: Normally weight decay won't be set as multiplier?

Comment (piiswrong) [inline python/mxnet/optimizer.py]: caffe has the same parameters lr_mult and wd_mult
